src/build/cargo.rs

```diff
@@ -105,9 +107,15 @@ fn run_cargo(compilation_cx: Arc<Mutex<CompilationContext>>,
 
     let ws = Workspace::new(&manifest_path, &config).expect("could not create cargo workspace");
 
+    // Create and store a build plan for the created Workspace
+    let build_plan = create_plan(&ws).unwrap();
+    compilation_cx_access.build_plan = Some(build_plan);
+    // Release the lock on the compilation_cx
+    mem::drop(compilation_cx_access);
```

Comment:
Could we use a scope rather than `mem::drop`? Or better, drop the lock and pick it up again here?